packages/core/test/actions.test.ts

```diff
@@ -415,6 +415,62 @@ describe('entry/exit actions', () => {
       expect(actual).toEqual(['loaded entry']);
     });
 
+    it("shouldn't use referenced custom action over a builtin one when there is a naming conflict", () => {
+      const spy = jest.fn();
+      const machine = createMachine(
+        {
+          context: {
+            assigned: false
+          },
+          on: {
+            EV: {
+              actions: assign({ assigned: true })
+            }
+          }
+        },
+        {
+          actions: {
+            'xstate.assign': spy
+          }
+        }
+      );
+
+      const actor = interpret(machine).start();
+      actor.send({ type: 'EV' });
+
+      expect(spy).not.toHaveBeenCalled();
+      expect(actor.getSnapshot().context.assigned).toBe(true);
+    });
+
+    it("shouldn't use referenced custom action over a builtin one when there is a naming conflict", () => {
```

Comment:
```suggestion
    it("shouldn't use inline named custom action over a builtin one when there is a naming conflict", () => {
```

Reply:
ah, copy-pasta from the other added test case went wrong 😅